xbmc/video/VideoInfoScanner.cpp

```diff
@@ -429,8 +429,12 @@ namespace VIDEO
       if (!info2) // skip
         continue;
 
+      // Discard all .nomedia folders
+      if (pItem->m_bIsFolder && HasNoMedia(pItem->GetPath()))
```

Comment:
this changes the behaviour of .nomedia with respect to recursions. ie before you could have .nomedia in the stop strDirectory and we would stop scanning, now it looks for .nomedia in each of the subfolders.